src/clj/harja/palvelin/palvelut/yllapitokohteet.clj

```diff
@@ -502,25 +502,52 @@
                                                :yllapitokohde-id id
                                                :osat korjatut-kohdeosat})))))
 
+(defn- validoi-tallennettava-yllapitokohteet [db tallennettavat-kohteet vuosi]
+  (let [saman-vuoden-kohteet (q/hae-yhden-vuoden-yha-kohteet db {:vuosi vuosi})]
+    (reduce
+      (fn [virheet tallennettava-kohde]
+        (let [kohteen-virheet
+              (remove nil? (map (fn [verrattava-kohde]
+                                  (when (and (not= (:id tallennettava-kohde) (:id verrattava-kohde))
+                                             (tr/tr-vali-leikkaa-tr-valin? tallennettava-kohde verrattava-kohde))
+                                    {:validointivirhe :kohteet-paallekain
+                                     :kohteet [(select-keys tallennettava-kohde [:kohdenumero :nimi :urakka
+                                                                                 :tr-numero :tr-alkuosa :tr-alkuetaisyys
+                                                                                 :tr-loppuosa :tr-loppuetaisyys])
+                                               (select-keys verrattava-kohde [:kohdenumero :nimi :urakka
+                                                                              :tr-numero :tr-alkuosa :tr-alkuetaisyys
+                                                                              :tr-loppuosa :tr-loppuetaisyys])]}))
+                                saman-vuoden-kohteet))]
+          (if (empty? kohteen-virheet)
+            virheet
+            (concat virheet kohteen-virheet))))
+      []
+      tallennettavat-kohteet)))
+
 (defn tallenna-yllapitokohteet [db user {:keys [urakka-id sopimus-id vuosi kohteet]}]
```

Comment:
Tuli mieleen että tässähän pitäis olla ! nimessä. Ei nyt välttämättä lisätyön arvoinen juttu :D